File: app/src/main/java/com/garymcgowan/moviepedia/dagger/ApplicationComponent.java
package com.garymcgowan.moviepedia.dagger;

import com.garymcgowan.moviepedia.App;
import com.garymcgowan.moviepedia.view.details.MovieDetailsActivity;
import com.garymcgowan.moviepedia.view.search.MovieListActivity;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by Gary on 2016/11/02.
 */

@Singleton
@Component(modules = {ApplicationModule.class, NetworkModule.class})
public interface ApplicationComponent {
    void inject(App application);

    void inject(MovieDetailsActivity fragment);

    void inject(MovieListActivity activity);


}

File: app/src/main/java/com/garymcgowan/moviepedia/dagger/ApplicationComponent.kt
package com.garymcgowan.moviepedia.dagger

import com.garymcgowan.moviepedia.App
import com.garymcgowan.moviepedia.view.details.MovieDetailsActivity
import com.garymcgowan.moviepedia.view.search.MovieListActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(ApplicationModule::class, NetworkModule::class))
interface ApplicationComponent {
    fun inject(application: App)

    fun inject(fragment: MovieDetailsActivity)

    fun inject(activity: MovieListActivity)


}
